### PR TITLE
refactor: deploy scripts with type id

### PR DESCRIPTION
**Pull request by mkxbl**
1. deploy scripts with type id
2. add useful info to mint/burn log
3. increase mint tx retry times to 10

### CODE REVIEW

File: ckb-contracts/tests/src/eth_recipient_typescript.rs [hunk 1]
```diff
@@ -1,21 +1,22 @@
-use crate::utils::{case_builder::*, case_runner};
+// TODO capsule test do not support hash_type Byte(0x01) yet, so remove recipient tests temporally
```
Comment (huwenchao) [inline ckb-contracts/tests/src/eth_recipient_typescript.rs:1]: Is it possible to use code hash_type in test and dev, type hash_type in testnet and mainnet?

File: ckb-contracts/types/src/config.rs [hunk 2]
```diff
@@ -14,10 +14,9 @@ cfg_if::cfg_if! {
         pub const SUDT_HASH_TYPE: u8 = 1;
     } else if #[cfg(feature = "devnet")] {
         pub const SUDT_CODE_HASH: [u8; 32] = [
-            225, 227, 84, 214, 214, 67, 173, 66, 114, 77, 64, 150, 126, 51, 73, 132, 83, 78, 3, 103,
-            64, 92, 90, 228, 42, 157, 125, 99, 215, 125, 244, 25,
+            19, 19, 160, 234, 165, 113, 169, 22, 142, 68, 206, 186, 26, 13, 10, 50, 136, 64, 217, 222, 67, 170, 178, 56, 138, 247, 200, 96, 181, 124, 154, 12,
```
Comment (huwenchao) [inline ckb-contracts/types/src/config.rs:17]: how Is it fixed in devnet?

File: ckb-contracts/tests/src/eth_recipient_typescript.rs (hunk 1 above)
Comment (mkxbl) [reply ckb-contracts/tests/src/eth_recipient_typescript.rs:1]: Yes, i will use conditional compile just like SUDT script

File: ckb-contracts/types/src/config.rs (hunk 2 above)
Comment (mkxbl) [reply ckb-contracts/types/src/config.rs:17]: If scripts to be deployed fixed, it will be fixed. Otherwise, its value will change.

**Pull request merged**